packages/data-stores/src/plans/selectors.ts

```diff
@@ -131,26 +132,28 @@ export const getPlanByPath = (
 
 	return select( STORE_KEY )
 		.getSupportedPlans( locale )
-		.find( ( plan ) => plan.periodAgnosticSlug === planProduct?.periodAgnosticSlug );
+		.find( ( plan ) => plan.periodAgnosticSlug === planProduct.periodAgnosticSlug );
```

Comment:
The `?.` check wasn't necessary, because of the typeguard check a few lines above

Reply:
Nitpick mode: Scope creep?

Reply:
Reverted in https://github.com/Automattic/wp-calypso/pull/50492/commits/6fe2a1e0f32d57bdbee7ec12eaeed06e5489b1a2, and opened #50666 to include these changes in a separate PR.